src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -178,35 +194,80 @@ func (mod *mgmtModule) getAttachInfoResp(ctx context.Context, numaNode int, sys
 	return mod.attachInfo.Get(ctx, numaNode, sys, mod.getAttachInfoRemote)
 }
 
-func (mod *mgmtModule) getProviderAttachInfo(srvResp *mgmtpb.GetAttachInfoResp) (*mgmtpb.GetAttachInfoResp, error) {
-	if mod.provider == "" || mod.provider == srvResp.ClientNetHint.Provider {
+func (mod *mgmtModule) getInterfaceProviders(iface, domain string) common.StringSet {
+	if iface == "" {
+		return nil
+	}
+
+	if domain == "" {
+		domain = iface
+	}
+
+	fis, err := mod.fabricInfo.localNUMAFabric.FindDevice(&FabricIfaceParams{
+		Interface: iface,
+		Domain:    domain,
+	})
+	if err != nil {
+		mod.log.Errorf("client-requested fabric interface/domain not detected: %s", err.Error())
+		mod.log.Error("communications on this interface may fail")
+		return nil
+	}
+
+	providers := common.NewStringSet()
+	for _, fi := range fis {
+		providers.AddUnique(fi.Providers()...)
+	}
+	return providers
+}
+
+func (mod *mgmtModule) selectAttachInfo(srvResp *mgmtpb.GetAttachInfoResp, reqProviders common.StringSet) (*mgmtpb.GetAttachInfoResp, error) {
+	providers := reqProviders
+	if mod.provider != "" {
+		if len(reqProviders) > 0 && !reqProviders.Has(mod.provider) {
+			mod.log.Errorf("configured provider %q not included in requested interface's detected providers: %s", reqProviders)
```

Comment:
same as above, I'm guessing you just want the "communications on this interface may fail" to be an independent entry?

Reply:
Yes, the idea was that we want to make this somewhat noisy so that if there are communications failures due to invalid parameters supplied via env variables it's not a mystery. We don't want the env variables to be used as a primary configuration method for users, hence the relaxed approach to validation ("okay, if you think this will work we'll try it, but as far as we know it's busted").

Reply:
Yeah, my goal was to make it a bit loud. I wanted to clarify what the consequences of the error are in case someone comes looking.